app/services/auth/firebase_auth.py

```
@@ -0,0 +1,382 @@
+        UserNotFound: If user with given UID doesn't exist
+        InternalServerError: If database update fails
+    """
+    user = session.query(User).filter(User.firebase_uid == uid).first()
+    if not user:
+        raise UserNotFound(uid)
+    # Update user details here as needed.
+    try:
+        update_data = {k: v for k, v in user_updates.model_dump(exclude={'password'}).items() if v is not None}
+        session.query(Profile).filter(Profile.user_id == user.id).update(update_data)
+        session.commit()
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)